taskchampion/lib/src/replica.rs

```diff
@@ -408,20 +607,41 @@ pub unsafe extern "C" fn tc_replica_sync(
 
 #[ffizz_header::item]
 #[ffizz(order = 902)]
-/// Undo local operations until the most recent UndoPoint.
+/// Return undo local operations until the most recent UndoPoint.
+///
+/// ```c
+/// EXTERN_C TCReplicaOpList tc_replica_get_undo_ops(struct TCReplica *rep);
+/// ```
+#[no_mangle]
+pub unsafe extern "C" fn tc_replica_get_undo_ops(rep: *mut TCReplica) -> TCReplicaOpList {
+    wrap(
+        rep,
+        |rep| Ok(TCReplicaOpList::from(rep.get_undo_ops()?)),
+        TCReplicaOpList::from(Vec::new()),
+    )
```

Comment:
Use the PassByValue mechanics here, after mapping the item types:
```suggestion
    wrap(
        rep,
        |rep| {
            // SAFETY:
            //  - caller will free this value, either with tc_replica_commit_undo_ops or
            //  tc_replica_op_list_free.
            Ok(unsafe {
                TCReplicaOpList::return_val(
                    rep.get_undo_ops()?
                        .into_iter()
                        .map(TCReplicaOp::from)
                        .collect(),
                )
            })
        },
        // SAFETY:
        //  - caller will free this value, either with tc_replica_commit_undo_ops or
        //  tc_replica_op_list_free.
        unsafe { TCReplicaOpList::return_val(Vec::new()) },
    )
```
(That last unsafe bit could be replaced with a `Default` implementation for `TCReplicaOpList`)